### PR TITLE
Update to support new DB BahnPark API

### PR DESCRIPTION
**Pull request by dennispost**
- [x] implement new api schema
- [x] add tests

### CODE REVIEW

**Comment by dennispost** [comment]
@lightsprint09 https://github.com/dbsystel/1BahnQL/blob/feature/Update_Parkingspaces_for_new_API/test/Parkingspace/ParkingspaceLoaderMock.js#L18 Should we remove this filter logic from loader?

**Comment by lightsprint09** [comment]
Yes remove it

File: Parkingspace/Parkingspace.js [hunk 1]
```diff
@@ -1,67 +1,73 @@
 const fetch = require('node-fetch');
```
Comment (lightsprint09) [inline Parkingspace/Parkingspace.js]: this should not be needed

File: Parkingspace/Parkingspace.js [hunk 2]
```diff
@@ -1,67 +1,73 @@
...
+    this.openingHours = space.openingHours;
+    this.openingHoursEn = space.openingHoursEn;
+    this.numberParkingPlaces = space.numberParkingPlaces;
+    this.numberHandicapedPlaces = space.numberHandicapedPlaces;
+    this.isSpecialProductDb = space.isSpecialProductDb;
+    this.isOutOfService = space.isOutOfService;
+    this.type = space.type;
+
+    // space info
+    this.clearanceWidth =  access(space, 'spaceInfo.clearanceWidth');
```
Comment (lightsprint09) [inline Parkingspace/Parkingspace.js]: after `=`there are 2 spaces

File: Parkingspace/ParkingspaceLoader.js [hunk 3]
```diff
@@ -19,35 +20,24 @@ class ParkingspaceLoader {
...
-        if (result.count > 0) {
-          const filteredResult = result.results.filter(elem => elem.parkraumId == spaceId);
-
-          if (filteredResult.length > 0) {
-            return filteredResult[0];
-          }
-          return null;
-        }
-      });
+      .then(result => result);
```
Comment (lightsprint09) [inline Parkingspace/ParkingspaceLoader.js]: this is not needed

File: Parkingspace/ParkingspaceLoader.js [hunk 4]
```diff
@@ -74,7 +64,7 @@ class ParkingspaceLoader {
   }
 
   nearbyParkingspaces(latitude, longitude, radius) {
-    const url = `${baseURL}/sites`;
+    const url = `${baseURL}/spaces?limit=100`;
```
Comment (lightsprint09) [inline Parkingspace/ParkingspaceLoader.js]: why do we use 100 asa limit here while using 1000 for spacesAtStation

File: schema.js [hunk 5]
```diff
@@ -192,66 +192,63 @@ const schema = buildSchema(`
   }
 
   type ParkingSpace {
+    type: String
```
Comment (lightsprint09) [inline schema.js:195]: can type be an enum

File: schema.js [hunk 6]
```diff
@@ -192,66 +192,63 @@ const schema = buildSchema(`
...
+    address: MailingAddress
+    distance: String
+    facilityType: String
+    facilityTypeEn: String
+    openingHours: String
+    openingHoursEn: String
+    numberParkingPlaces: String
+    numberHandicapedPlaces: String
+    isSpecialProductDb: Boolean
+    isOutOfService: Boolean
```
Comment (lightsprint09) [inline schema.js]: Boolean should not be optional... I am not sure if this is possible here

File: test/Parkingspace/testParkingspaceService.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    parkingspaceLoaderMock.result = require('./SingleParkingspaceMockResult');
+    const promise = parkingspaceService.parkingspaceBySpaceId(100082);
+    return Promise.all([
+      expect(promise).to.eventually.have.property('id', 100082),
+      expect(promise).to.eventually.have.deep.property('address', { city: 'Frankfurt am Main', zipcode: '60329', street: 'Poststraße' }),     
+
+    ]);
+  });
+
+  it.skip('parkingspaceBySpaceId with wrong spaceID should return null', () => {
```
Comment (lightsprint09) [inline test/Parkingspace/testParkingspaceService.js]: what does `it.skip` does?

**Review by lightsprint09 — CHANGES_REQUESTED** [review]
Looks very good. I like the tests 👍

File: schema.js (hunk 5 above)
Comment (dennispost) [reply schema.js:195]: Maybe, but its not documented so I have no clue wich types do exist

File: test/Parkingspace/testParkingspaceService.js (hunk 7 above)
Comment (dennispost) [reply test/Parkingspace/testParkingspaceService.js]: it skips a test ;-)

**Pull request merged**